File: ZkTony-Api/db/src/system/models/software.rs
use serde::Deserialize;

#[derive(Deserialize, Debug, Clone)]
pub struct SoftwareSaveReq {
    pub id: String,
    pub package: String,
    pub version_code: i32,
    pub version_name: String,
    pub build_type: String,
    pub remarks: Option<String>,
    pub create_by: Option<String>,
    pub create_time: Option<String>,
}

#[derive(Deserialize, Debug, Clone)]
pub struct SoftwareDeleteReq {
    pub id: String,
}

#[derive(Deserialize, Debug, Clone)]
pub struct SoftwareGetReq {
    pub id: Option<String>,
    pub package: Option<String>,
    pub build_type: Option<String>,
}

File: ZkTony-Manager/app/src/main/kotlin/com/zktony/manager/data/remote/model/Software.kt
package com.zktony.manager.data.remote.model

import com.zktony.www.common.extension.currentTime

/**
 * @author: 刘贺贺
 * @date: 2023-02-16 13:16
 */
data class Software(
    val id: String = "",
    val `package`: String = "com.zktony.manager",
    val version_name: String = "1.0.0",
    val version_code: Int = 1,
    val build_type: String = "debug",
    val remarks: String = "",
    val create_by: String = "",
    val create_time: String = currentTime(),
)